src/cache/local.ml

```diff
@@ -140,9 +140,14 @@ module Metadata_file = struct
     >>= of_sexp
 end
 
-let path_files cache = Path.relative cache.root "files"
+let path_ cache d =
+  let res = Path.relative cache.root d in
+  Path.mkdir_p res;
+  res
 
-let path_meta cache = Path.relative cache.root "meta"
+let path_files cache = path_ cache "files"
```

Comment:
Doing a system call every-time in such a function seems not very efficient. Could it be done at some initialisation stage?